### PR TITLE
YLP-1903 Timezone detection should exclude some data types

### PR DESCRIPTION
**Pull request by CedricPerotto**

### CODE REVIEW

**Comment by fatb38** [comment]
Could cherry pick data-api tests in PR #935

File: packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts [hunk 1]
```diff
@@ -126,6 +126,7 @@ class MedicalDataService {
         } else {
           message = String(error)
         }
+        console.log(raw)
```
Comment (TimCanu) [inline packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts:129]: To be removed

File: packages/medical-domain/test/MedicalDataService.test.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+  const rangeEnd = new Date(medicalData.endpoints[1])
+  expect(rangeEnd.valueOf()).toBeGreaterThan(rangeStart.valueOf())
+  const allData = medicalData.data.filter(d => !['fill', 'pumpSettings', 'upload'].includes(d.type))
+  const outOfRangeData = allData.filter(
+    (dat) => {
+      const epoch = 'epochEnd' in dat ? dat.epochEnd : dat.epoch
+      return epoch < rangeStart.valueOf() || epoch > rangeEnd.valueOf()
+    }
+  )
+  if (outOfRangeData.length > 0) {
```
Comment (TimCanu) [inline packages/medical-domain/test/MedicalDataService.test.ts:150]: Do we still want to keep this code?

File: packages/viz/test/utils/print/data.test.js [hunk 3]
```diff
@@ -156,8 +156,11 @@ describe('print data utils', () => {
       const startDate = moment.tz('2019-05-20', 'Europe/Paris').locale('fr').startOf('week')
       const endDate = moment.tz('2019-05-20', 'Europe/Paris').locale('fr').endOf('week')
       filtered = selectDailyViewData(tidelineData, startDate, endDate)
+      console.log(filtered)
```
Comment (TimCanu) [inline packages/viz/test/utils/print/data.test.js:159]: Remove the three console log

File: packages/yourloops/test/unit/lib/data/data-api.test.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2021, Diabeloop
```
Comment (TimCanu) [inline packages/yourloops/test/unit/lib/data/data-api.test.tsx:2]: 2022

**Review by TimCanu — COMMENTED** [review]
I don't understand luch of this... But if you fix my comments, create a BETA image and run it on yourloops CI, then it is good for me

File: packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts (hunk 1 above)
Comment (CedricPerotto) [reply packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts:129]: logging only error doesn't give many context... I'll change the log to include both error and data (with only one console.log).

**Comment by CedricPerotto** [comment]
> I don't understand much of this... But if you fix my comments, create a BETA image and run it on yourloops CI, then it is good for me
beta18 released and tested in yourloops here: 
https://github.com/mdblp/yourloops/pull/791

**Review by TimCanu — APPROVED** [review]

**Pull request merged**